python/cudf/cudf/tests/test_applymap.py

```diff
@@ -58,3 +58,46 @@ def test_applymap_change_out_dtype():
     expect = np.array(data, dtype=float)
     got = out.to_numpy()
     np.testing.assert_array_equal(expect, got)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [1, 2, 3], "b": [4, 5, 6]},
+        {"a": [1, 2, 3], "b": [1.0, 2.0, 3.0]},
+        {"a": [1, 2, 3], "b": [True, False, True]},
+        {"a": [1, NA, 2], "b": [NA, 4, NA]},
+    ],
+)
+@pytest.mark.parametrize(
+    "func",
+    [
+        lambda x: x + 1,
+        lambda x: x - 1,
+        lambda x: x + 0.5,
```

Comment:
Maybe we could reduce the number of tests in this matrix while retaining similar coverage:
```suggestion
        lambda x: x + 1,
        lambda x: x - 0.5,
```